src/models/methods/subscribe.ts

```diff
@@ -0,0 +1,107 @@
+import { BaseRequest, BaseResponse } from './baseMethod'
+import { Currency, StreamType } from "../common"
+import { TransactionMetadata } from '../common/transaction'
+
+interface Book {
+    taker_gets: Currency
+    taker_pays: Currency
+    taker: string
+    snapshot?: boolean
+    both?: boolean
+}
+  
+export interface SubscribeRequest extends BaseRequest {
+    command: "subscribe"
+    streams?: StreamType[]
+    accounts?: string[]
+    accounts_proposed?: string[]
+    books?: Book[]
+    url?: string
+    url_username?: string
+    url_password?: string
+}
+
+export interface SubscribeResponse extends BaseResponse {
+    result: any
+}
+
+interface BaseStream {
+    type: string
+}
+
+export interface LedgerStream extends BaseStream {
+    type: "ledgerClosed"
+    fee_base: number
+    fee_ref: number
+    ledger_hash: string
+    ledger_index: number
+    ledger_time: number
+    reserve_base: number
+    reserve_inc: number
+    txn_count: number
+    validated_ledgers?: string
+}
+
+export interface ValidationStream extends BaseStream {
+    type: "validationReceived"
+    amendments?: string[]
+    base_fee?: number
+    flags: number
+    full: boolean
+    ledger_hash: string
+    ledger_index: string
+    load_fee?: number
+    master_key?: string
+    reserve_base?: number
+    reserve_inc?: number
+    signature: string
+    signing_time: number
+    validation_public_key: string
+}
+
+export interface TransactionStream extends BaseStream {
```

Comment:
There seems to be a hidden additional TransactionStream called "transactions_proposed stream" which is very similar to TransactionStream, except it has a couple different values (Ex. validated is "optional in transactions_proposed stream", but mandatory in the "transactions stream") - The details for the proposed stream are described exclusively in paragraph form here: https://xrpl.org/subscribe.html#transaction-streams

That paragraph also describes an "accounts_proposed stream" which is similarly loosely documented in paragraph form.

Reply:
@mvadari Do you think it makes sense to add a TransactionProposedStream interface, or is that outside the scope?

Reply:
Decision: Not worth it to make custom interfaces, instead just made these interfaces accommodate the "proposed" formats as well since it's just small tweaks.